### DISCUSSION TITLE
Finding company/developer to make plugin

### DISCUSSION THREAD

**Discussion by ledj** [haraka/Haraka#3049] [General]
Hi

I'm wondering where to find and hire a company or developer for making a plugin for Haraka?

**Comment by msimerson** [comment]
Post a description of what you want here. There are Haraka devs (I'm among them) who occasionally have time for side projects, especially when the side project(s) overlaps with something they've been itching to do.

**Reply by ledj** [reply]
Ok, great.
In short, we're using Haraka outbound only with sendy.co (self-hosted newsletter service), and have a setup running already working fine. EU law (GDPR) prohibits us (and our customers) from using MailChimp etc.
The challenge of using a self-hosted newsletter service is maintaining the reputation of our IP addresses sending mail.
To keep mailing lists clean from non-existent addresses and other delivery fail scenarios, we need a plugin that can register if a newsletter mail has bounced or not.
It would be great with a plugin that can save the delivery status somewhere so we can request that "database" and remove an address in Sendy if it has failed for some reason.

**Reply by thenitai** [reply]
Hey Lars,

Take a look at the plugin that we use at Helpmonks ( https://helpmonks.com ). We use Haraka for all incoming emails (millions a day) and for outgoing (though we are moving away from that).

Everything is stored in the database and you can do whatever you like with that information. It does so much more too with parsing, storing outgoing emails, etc.

https://github.com/Helpmonks/haraka-plugin-mongodb

**Reply by msimerson** [reply]
It looks like the mongodb plugin may do what you want. The other option I would suggest is extending the [elasticsearch](https://github.com/haraka/haraka-plugin-elasticsearch) plugin. The ES plugin is a more tailored towards logging SMTP transactions but it doesn't (yet) have outbound logging. Does your organization have experience with mongo or ES?

**Reply by ledj** [reply]
Yes, mongodb plugin looks promising, we'll try it out. We have experience with ES and used it for logging (webserver access log) for a while, but changed to influxdb which is more lightweight for our purpose. ES used a lot of memory (several GiB) and CPU, the same load on influxdb uses only ~540MiB (+ fluentbit that uses ~32MiB of memory) and fits our container-based environment better.

**Reply by msimerson** [reply]
They are different tools for scratching different itches. Logging the disposition of millions of emails along with deliverability metrics seems like a good fit for influx. Doing more, like being able to answer, "how many hammy emails has email@example.com received from sweden in the last 3 years, where rspamd suggested "reject," would be a stretch for all but elasticsearch. It's true that ES has rather high resource requirements, but the 12 year old server I just retired has 192GB of RAM. Its 7 year old replacement is 1/8 full with "only" 256GB of RAM. It costs more to use a less optimal tool than to buy a stick of RAM.

**Reply by ledj** [reply]
Just a quick follow-up, the mongodb plugin works great, and we can remove mail addresses that bounce in Sendy (sendy.co).
